packages/core/useWorkerWebFunction/index.ts

```diff
@@ -0,0 +1,112 @@
+/* this implementation is a vue port of https://github.com/alewin/useWorker by Alessio Koci */
+
+import { ref, onMounted, onUnmounted, Ref } from '../../api'
+import WORKER_STATUS from './lib/status'
+import createWorkerBlobUrl from './lib/createWorkerBlobUrl'
+
+type Options = {
+  timeout?: number
+  dependencies?: string[]
+}
+const PROMISE_RESOLVE = 'resolve'
+const PROMISE_REJECT = 'reject'
+const DEFAULT_OPTIONS: Options = {
+  timeout: undefined,
+  dependencies: [],
+}
+
+/* eslint-disable arrow-parens */
+export const useWebWorkerFunction = <T extends (...fnArgs: any[]) => any>(
+  fn: T, options: Options = DEFAULT_OPTIONS,
+) => {
+  /* eslint-enable arrow-parens */
+  const worker: Ref<Worker & { _url?: string } | undefined> = ref(undefined)
+
+  const workerStatus: Ref<WORKER_STATUS> = ref(WORKER_STATUS.PENDING)
+  const promise: Ref<{ [PROMISE_REJECT]?: (result: ReturnType<T> | ErrorEvent) => void;[PROMISE_RESOLVE]?: (result: ReturnType<T>) => void }> = ref({})
+  const timeoutId: Ref<number | undefined> = ref(undefined)
+
+  const killWorker = (status = WORKER_STATUS.PENDING) => {
+    if (worker.value && worker.value._url) {
+      worker.value.terminate()
+      URL.revokeObjectURL(worker.value._url)
+      promise.value = {}
+      worker.value = undefined
+      window.clearTimeout(timeoutId.value)
+      workerStatus.value = status
+    }
+  }
+
+  onMounted(() => {
+    killWorker()
+  })
+
+  onUnmounted(() => {
+    killWorker()
+  })
+
+  const generateWorker = () => {
+    const {
+      dependencies = DEFAULT_OPTIONS.dependencies,
+      timeout = DEFAULT_OPTIONS.timeout,
+    } = options
+    const blobUrl = createWorkerBlobUrl(fn, dependencies!)
```

Comment:
```suggestion
    const {
      dependencies = DEFAULT_OPTIONS.dependencies || [],
      timeout = DEFAULT_OPTIONS.timeout,
    } = options
    const blobUrl = createWorkerBlobUrl(fn, dependencies)
```

I would not suggest using null assertion, maybe this would be better.